tests/calc/classes/rparams/special/test_max_tl_displacement.py

```diff
@@ -0,0 +1,123 @@
+        'no value': (1.1, NO_VALUE)
+        }
+    valid_float = {
+        'one': (0.15,),
+        'two': (0.15, 0.2),
+        }
+    valid_refcalc = {
+        'ignore': (('ignore',), NO_VALUE),
+        'no duration': (('refcalc',), NO_VALUE),
+        'pure float': (('refcalc', '600'), 600),
+        'pure float, lt': (('refcalc', '<', '600'), 600),
+        'seconds': (('refcalc', '30s'), 30),
+        'minutes': (('refcalc', '5m'), 300),
+        'hours': (('refcalc', '2h'), 7200),
+        }
+
+    def test_valid_initialization(self):
+        obj = MaxTLDisplacement(geo=0.5, _vib=0.3)
+        assert isinstance(obj, MaxTLDisplacement)
+        assert obj.geo == 0.5
+        assert obj.vib == 0.3
+
+    def test_default_vib_fallback_to_geo(self):
+        obj = MaxTLDisplacement(geo=0.8)
+        assert obj.vib == 0.8
+
+    @parametrize('geo,vib', valid_creation.values(), ids=valid_creation)
+    def test_type_conversion_and_range(self, geo, vib):
+        obj = MaxTLDisplacement(geo=geo, _vib=vib)
+        assert isinstance(obj.geo, float)
+        if vib is not NO_VALUE:
+            assert isinstance(obj._vib, float)
+
+    @parametrize(args=valid_float.values(), ids=valid_float)
+    def test_from_value(self, args):
+        obj = MaxTLDisplacement.from_value(args)
+        assert isinstance(obj, MaxTLDisplacement)
+        assert obj.geo == args[0]
+        if len(args) > 1:
+            assert obj._vib == args[1]
+
+    @parametrize(args=valid_float.values(), ids=valid_float)
+    def test_assign_float_values(self, args):
+        obj = MaxTLDisplacement(geo=0.5)
+        obj.assign_float_values(args)
+        assert obj.geo == args[0]
+        if len(args) > 1:
+            assert obj._vib == args[1]
+
+    def test_assign_named_geo(self):
+        obj = MaxTLDisplacement(geo=1.0)
+        obj.assign_single_value('geo', 0.7)
+        assert obj.geo == 0.7
+
+    def test_assign_named_vib(self):
+        obj = MaxTLDisplacement(geo=1.0)
+        obj.assign_single_value('vib', 0.6)
+        assert obj._vib == 0.6
+
+    @parametrize('values,expect', valid_refcalc.values(), ids=valid_refcalc)
+    def test_assign_refcalc(self, values, expect):
+        obj = MaxTLDisplacement(geo=1.0)
+        obj.assign_action(values)
+        assert obj.action == values[0]
+        if obj.action == 'refcalc':
+            assert obj.max_duration == expect
+
+
+class TestMaxTLDisplacementInvalid:
+    """Collection of tests for invalid inputs to MaxTLDisplacement objects."""
+    invalid = {
+        'not numeric': ('a', TypeError),
+        'negative': (-1, ValueError),
+        'zero': (0, ValueError),
+        }
+
+    @parametrize('value,err', invalid.values(), ids=invalid)
+    def test_invalid(self, value, err):
+        with pytest.raises(err):
+            MaxTLDisplacement(value)
+
+
+class TestMaxTLDisplacementTooFar:
+    """Collection of tests for whether thresholds have been reached in
+    MaxTLDisplacement objects."""
```

Comment:
```suggestion
    """Tests for whether displacement thresholds have been reached."""
```

(First, or only, line in docstring fits the 80 chars limit.)